connexion/security/aiohttp_security_handlers_factory.py

```diff
@@ -0,0 +1,111 @@
+import asyncio
+import functools
+import logging
+
+import aiohttp
+
+from ..exceptions import OAuthProblem, OAuthResponseProblem, OAuthScopeProblem
+from .security_handler_factory import SecurityHandlerFactory
+
+logger = logging.getLogger('connexion.api.security')
+
+
+class AioHttpSecurityHandlerFactory(SecurityHandlerFactory):
+    def __init__(self, pass_context_arg_name):
+        SecurityHandlerFactory.__init__(self, pass_context_arg_name)
+        self.client_session = None
+
+    def _generic_check(self, func, exception_msg):
+        need_to_add_context, need_to_add_required_scopes = self._need_to_add_context_or_scopes(func)
+
+        @asyncio.coroutine
+        def wrapper(request, *args, required_scopes=None):
+            kwargs = {}
+            if need_to_add_context:
+                kwargs[self.pass_context_arg_name] = request.context
+            if need_to_add_required_scopes:
+                kwargs[self.required_scopes_kw] = required_scopes
+            token_info = func(*args, **kwargs)
+            while asyncio.iscoroutine(token_info):
+                token_info = yield from token_info
+            if token_info is self.no_value:
+                return self.no_value
+            if token_info is None:
+                raise OAuthResponseProblem(description=exception_msg)
```

Comment:
There is no oauth server response here, and on further inspection this function is not just for oauth so raising `OAuthResponseProblem` is actually not correct. This is doing the same as the old security.py version, but that still does not make sense.